src/FileIndexer.ts

```diff
@@ -397,6 +444,84 @@ export class FileIndexer {
     }
     return node.getText() + ': ' + type()
   }
+
+  // Invokes the `onParent` callback for all "parents" of the provided node,
+  // where "parent" is loosely defined as the superclass or superinterface of
+  // that node. The callback is invoked on the `node` parameter itself if it's
+  // class-like or an interface.
+  private forEachParent(
```

Comment:
nit: Should this be called `forEachAncestor` instead? (If so, please rename the argument + update the doc comment above) IIUC that would be more accurate; using "parent" seems confusing because that seems to imply that the traversal will only go 1 level deep.